### PR TITLE
Fix logging

### PR DESCRIPTION
**Pull request by gavofyork**
- Fixes problematic colour-codes in the web interface.
- `--log-file` for redirecting logging to a file.

### CODE REVIEW

File: parity/setup_log.rs [hunk 1]
```diff
@@ -45,18 +51,48 @@ pub fn setup_log(init: &Option<String>, enable_color: bool) -> Arc<RotatingLogge
...
-			format!("{}{}:{}: {}", timestamp, record.level(), record.target(), record.args())
+			format!("{}{}:{}: {}", timestamp.apply(Colour::Black.bold()), record.level(), record.target(), record.args())
 		};
-		logger.append(format.clone());
+
+		let removed_color = kill_color(format.as_ref());
+		if let &Some(ref f) = &file {
+			// ignore errors - there's nothing we can do
+			let _ = f.locked().write_all(removed_color.as_bytes());
+			let _ = f.locked().write_all(b"\n");
```
Comment (rphmeier) [inline parity/setup_log.rs]: why lock twice?

**Comment by rphmeier** [comment]
lgtm other than mutex grumble

**Comment by arkpar** [comment]
Would be nice to disable colours for non-terminal output. Can be checked with `libc::isatty`

**Comment by gavofyork** [comment]
> Would be nice to disable colours for non-terminal output

another PR i think

**Comment by rphmeier** [comment]
still unnecessarily uses a mutex. i have a commit which fixes this at https://github.com/rphmeier/parity/commit/733644263cb29e52088265ab3b673d3633da8376

**Pull request merged**